### PR TITLE
Support middle-tier authentication strategy

### PR DESCRIPTION
**Pull request by jinanisha**
## How did you test this change?

- [x] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Comment by akphi** [comment]
@jinanisha is this ready for review?

**Comment by akphi** [comment]
@epsstan @jinanisha can you guys confirm the purpose of this PR, else we can close it.

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 1]
```diff
@@ -1015,6 +1017,30 @@ const UsernamePasswordAuthenticationStrategyEditor = observer(
...
 
+const MiddleTierUsernamePasswordAuthenticationStrategyEditor = observer(
+  (props: {
+    authSpec: MiddleTierUsernamePasswordAuthenticationStrategy;
+    isReadOnly: boolean;
+  }) => {
+    const { authSpec, isReadOnly } = props;
+    return (
+      <>
+        <ConnectionEditor_StringEditor
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:1028]: Would recommend adding a description for this input to guide better the users on what exactly this property does

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
Let's add a changeset to fix the build

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx (hunk 1 above)
Comment (jinanisha) [reply packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:1028]: Added in this [commit](https://github.com/finos/legend-studio/pull/1238/commits/b8abe20c271749242cc1317bbe47b2d36624b939)

**Comment by akphi** [comment]
@epsstan please don't add `legend-stack` label to ours for now. We'll find a label to indicate a change needs backend change.

File: .changeset/tiny-jeans-reflect.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-manual-tests': patch
+---
+
+Add tests
```
Comment (MauricioUyaguari) [inline .changeset/tiny-jeans-reflect.md:5]: this changeset should be empty.

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx [hunk 3]
```diff
@@ -1014,6 +1016,30 @@ const UsernamePasswordAuthenticationStrategyEditor = observer(
...
+          isReadOnly={isReadOnly}
+          value={authSpec.vaultReference}
+          propertyName={'vault reference'}
+          update={(value: string | undefined): void =>
```
Comment (MauricioUyaguari) [inline packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:1032]: should we provide a description for this component so the user knows what this property is for?

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

File: packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx (hunk 3 above)
Comment (jinanisha) [reply packages/legend-studio/src/components/editor/edit-panel/connection-editor/RelationalDatabaseConnectionEditor.tsx:1033]: Added in this [commit](https://github.com/finos/legend-studio/pull/1238/commits/33190594ae714703d426ac7198aebfa73fed762d)

File: .changeset/tiny-jeans-reflect.md (hunk 2 above)
Comment (jinanisha) [reply .changeset/tiny-jeans-reflect.md]: Added correct changeset in this [commit](https://github.com/finos/legend-studio/pull/1238/commits/c8be5fe540dfc421847415c2a35aee08beedb737)

**Review by MauricioUyaguari — DISMISSED** [review]

File: .changeset/curly-houses-act.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@finos/legend-graph': patch
+'@finos/legend-studio': patch
```
Comment (MauricioUyaguari) [inline .changeset/curly-houses-act.md:3]:
````suggestion
'@finos/legend-application-studio': patch
````

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
Your roundtrip tests are failing. Please take a look

**Comment by MauricioUyaguari** [comment]
@jinanisha test is still failing

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**